qa/integration-tests/src/test/java/io/camunda/zeebe/broker/it/clustering/ClusteredSnapshotTest.java

```diff
@@ -0,0 +1,392 @@
+
+import io.camunda.zeebe.broker.Broker;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.configuration.DataCfg;
+import io.camunda.zeebe.broker.system.configuration.ExporterCfg;
+import io.camunda.zeebe.client.api.response.BrokerInfo;
+import io.camunda.zeebe.exporter.api.Exporter;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Context.RecordFilter;
+import io.camunda.zeebe.exporter.api.context.Controller;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.snapshots.SnapshotId;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.springframework.util.unit.DataSize;
+
+@RunWith(Parameterized.class)
+public class ClusteredSnapshotTest {
+
+  private static final Duration SNAPSHOT_INTERVAL = Duration.ofMinutes(5);
+
+  @Rule
+  public final ClusteringRule clusteringRule = new ClusteringRule(1, 3, 3, this::configureBroker);
+
+  @Parameter(0)
+  public Consumer<ClusteringRule> snapshotTrigger;
+
+  @Parameter(1)
+  public String description;
+
+  @Parameters(name = "{index}: {1}")
+  public static Object[][] snapshotTriggers() {
+    return new Object[][] {
+      new Object[] {
+        (Consumer<ClusteringRule>) ClusteringRule::triggerAndWaitForSnapshots,
+        "explicit trigger snapshot"
+      },
+      new Object[] {
+        (Consumer<ClusteringRule>) (rule) -> rule.getClock().addTime(SNAPSHOT_INTERVAL),
+        "implicit snapshot by advancing the clock"
+      }
+    };
+  }
+
+  @After
+  public void cleanUp() {
+    ControllableExporter.updatePosition(true);
+    ControllableExporter.EXPORTED_RECORDS.set(0);
+    ControllableExporter.RECORD_TYPE_FILTER.set(r -> true);
+    ControllableExporter.VALUE_TYPE_FILTER.set(r -> true);
+  }
+
+  @Test
+  public void shouldTakeSnapshotsOnAllNodes() {
+    // given
+    ControllableExporter.updatePosition(true);
+
+    publishMessages();
+    ControllableExporter.updatePosition(false);
+    publishMessages();
+
+    // when - then
+    awaitUntilAsserted(
+        (broker) -> {
+          triggerSnapshotRoutine();
+          assertThat(broker).havingSnapshot();
+        });
```

Comment:
In 1.1 only leader take snapshots. Leader then replicates the snapshot to followers. So here it would be enough to take snapshot on leader and wait for it on the leader. (There are already other tests for replication).  But it is not a problem to triggerSnapshotRoutine on the followers. It is just that it has no effect.

Reply:
Changed the `#awaitUntilAsserted()` to assert only the leader. I kept the rest as it is.